Update Nix and CI to use Ruby 3.1

kevin-j-m opened the pull request:
This sets up local development to use ruby 3.1.4 in nix.

Nix is updated to nixpkgs-23.05 to support this.

The Circle CI configuration is also updated to run ruby 3.1.

kevin-j-m committed:
Update Nix and CI to use Ruby 3.1

This sets up local development to use ruby 3.1.4 in nix.

Nix is updated to nixpkgs-23.05 to support this.

The Circle CI configuration is also updated to run ruby 3.1.

kevin-j-m commented on circle.yml:
```diff
@@ -1,5 +1,5 @@
 machine:
   ruby:
-    version: 2.3.4
+    version: 3.1.4
```
It looks like this is still set up to use Circle 1.0. Would we like to update to 2.0? Or have CI running in some different way?

kohder replied:
We don't have CI hooked up for this project. It's so small and infrequently changed that it's okay to say the suite should be run by the dev when making a change.
Were you in here making another change and decided to do these upgrades?

kevin-j-m replied:
From the description in #7 
> Looks like there are more changes coming in Sidekiq 7 but I think they will only break the test suite

I'm planning on getting the test suite working with Sidekiq 7. And noticed the ruby version when I was getting started.

Should we remove the circle file entirely then?

kohder replied:
Turned out to work with 7 just fine. I recently blessed it with Hedwig's upgrade to Sidekiq 7 -- I updated the README.
Totally up to you which way you want to go with CI in here.

kevin-j-m replied:
> Turned out to work with 7 just fine.

My local test suite for this repo isn't working in Sidekiq 7. The plan is to get that passing.

kevin-j-m committed:
Remove Circle config

This repo is not using Circle CI, so this removes the configuration to
eliminate any confusion about that.

kevin-j-m commented:
I added f99df7d8974967b229c6048e11305b1244e0490e to drop the Circle config. I'll handle CI and the test suite in Sidekiq 7 in separate PRs.

This should be good for review/delivery (once approved) independently.

kohder reviewed (approved):
Thanks!
